server/src/main/java/org/elasticsearch/transport/RemoteClusterService.java

```diff
@@ -351,10 +361,13 @@ public RemoteClusterConnection getRemoteClusterConnection(String cluster) {
     @Override
     public void listenForUpdates(ClusterSettings clusterSettings) {
         super.listenForUpdates(clusterSettings);
-        clusterSettings.addAffixUpdateConsumer(REMOTE_CLUSTER_SKIP_UNAVAILABLE, this::updateSkipUnavailable, (alias, value) -> {});
+        if (isStateless == false) {
+            clusterSettings.addAffixUpdateConsumer(REMOTE_CLUSTER_SKIP_UNAVAILABLE, this::updateSkipUnavailable, (alias, value) -> {});
+        }
```

Comment:
I think this conditional registration also suggests that we don't want the setting to be configured at all.